### PR TITLE
[`RFC 3086`] Attempt to try to resolve blocking concerns  

### PR DESCRIPTION
**Pull request by c410-f3r**
Implements what is described at https://github.com/rust-lang/rust/issues/83527#issuecomment-1744822345 to hopefully make some progress.

It is unknown if such approach is or isn't desired due to the lack of further feedback, as such, it is probably best to nominate this PR to the official entities.

@rustbot labels +I-compiler-nominated

### CODE REVIEW

**Comment by Nilstrieb** [comment]
I don't think the compiler team is the right target for this. You should nominate it for the lang team instead.

**Comment by apiraino** [comment]
Following [this suggestion](https://github.com/rust-lang/rust/pull/117050#issuecomment-1774184258), I'll switch the nomination flags

@rustbot label -I-compiler-nominated +I-lang-nominated

**Comment by petrochenkov** [comment]
It's not clear from the PR description what is changed.
This PR only makes the `ignore(var)` -> `ignore($var)` change, but not the index order change.

**Comment by c410-f3r** [comment]
@petrochenkov Sorry, I didn't have the time to implement the index stuff. This will probably be done in the next few days.

File: tests/ui/macros/rfc-3086-metavar-expr/syntax-errors.rs [hunk 1]
```diff
@@ -125,14 +118,16 @@ macro_rules! open_brackets_without_tokens {
...
 }
 
 macro_rules! unknown_ignore_ident {
     ( $( $i:ident )* ) => {
         ${ignore(bar)}
-        //~^ ERROR variable `bar` is not recognized in meta-variable expression
+        //~^ ERROR meta-variable expressions must be referenced using a dollar sign
+        //~| ERROR expected expression
     };
 }
```
Comment (markbt) [inline tests/ui/macros/rfc-3086-metavar-expr/syntax-errors.rs:133]: These tests (which are for "unknown" values) should probably be updated to refer to `$foo` and `$bar` and then additional tests added for the `meta-variable expressions must be referenced using a dollar sign` case.

nit: The error message doesn't seem quite right to me.  The "expression" is `count(_)`, which doesn't need a `$`, but rather it is the *parameter* to the count or ignore method that needs it.  Perhaps `meta-variables within meta-variable expressions must be referenced using a dollar sign`, if that's not too long.

**Comment by nikomatsakis** [comment]
We discussed this in the @rust-lang/lang meeting — I don't see a comment and can't remember who was going to leave one so let me summarize meeting consensus.

TL;DR, given that @markbt agrees, we think you should land this PR, and we don't see a need for an FCP. However, please update the "OP" in the tracking issue to indicate the changes here and link to your justifications.

Why no FCP? This is not a 1-way door and our feeling was that there would not be strong opposition, given that folks driving the issue are in favor.

**Review by CohenArthur — APPROVED** [review]
this looks good to me :) thank you for getting this going again

**Comment by c410-f3r** [comment]
Thank you lang team for discussing this feature.

@petrochenkov Code is ready for review

For anyone who has permission, please update the tracking issue according to what @nikomatsakis described at https://github.com/rust-lang/rust/pull/117050#issuecomment-1802336843.

File: tests/ui/macros/rfc-3086-metavar-expr/syntax-errors.rs (hunk 1 above)
Comment (c410-f3r) [reply tests/ui/macros/rfc-3086-metavar-expr/syntax-errors.rs:133]: Thank you @markbt. The `meta-variable expressions must ...` message has been changed but if possible, I would like to perform further UI modifications in a following PR.

**Comment by c410-f3r** [comment]
> error parsing HTTP 429 response body: invalid character 'T' looking for beginning of value: "Too Many Requests (HAP429).\n"

Please `r` again
